cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -185,8 +186,8 @@ public BodyConsumer create(HttpRequest request, HttpResponder responder,
                              @PathParam("namespace-id") final String namespaceId,
                              @PathParam("app-id") final String appId)
     throws BadRequestException, NamespaceNotFoundException, AccessException {
-
-    ApplicationId applicationId = validateApplicationId(namespaceId, appId);
+    String versionId = RunIds.generate().getId();
```

Comment:
We should unify all the RunIds generation inside the `deployApp` method in the service. The `deployApp` methods in the ApplicationLifecycleService shouldn't take appVersion as a parameter.

Reply:
created a [JIRA](https://cdap.atlassian.net/browse/CDAP-19828), and added a TODO to refactor this later